services/spar/test-integration/Util/Core.hs

```diff
@@ -205,6 +209,17 @@ pendingWith :: (HasCallStack, MonadIO m) => String -> m ()
 pendingWith = liftIO . Test.Hspec.pendingWith
 
 
+-- | Run a probe several times, until a "good" value materializes or until patience runs out.
+-- The result may be good or not, depending on whether we run out of time.
```

Comment:
Tbh this function is still unintuitive enough for me that I feel it would be worth it to be more explicit. Here's my attempt:

```suggestion
-- If all retries were unsuccessful, 'aFewTimes' will return the last obtained value, even
-- if it does not satisfy the predicate.
```